### PR TITLE
Nicer error if loanbook lacks sector or borderline

### PR DESCRIPTION
**Pull request by maurolepore**

### CODE REVIEW

File: tests/testthat/test-match_name.R [hunk 1]
```diff
@@ -636,15 +636,30 @@ test_that("with relevant options allows loanbook with reserved columns", {
...
+    match_name(lacks_borderline, fake_ald()),
+    "Must have both `sector` and `borderline`"
+  )
+
+  lacks_sector <- mutate(fake_lbk(), borderline = TRUE)
+  expect_error(
+    match_name(lacks_sector, fake_ald()),
+    "Must have both `sector` and `borderline`"
+  )
 })
```
Comment (maurolepore) [inline tests/testthat/test-match_name.R:665]: These two tests now more clearly express the two issues in #330.

File: R/restructure_ald.R [hunk 2]
```diff
@@ -73,7 +73,11 @@ restructure_loanbook <- function(data, overwrite = NULL) {
 }
 
 may_add_sector_and_borderline <- function(data) {
-  if (has_sector_and_borderline(data)) {
+  if (lacks_borderline(data) || lacks_sector(data)) {
+    abort("Must have both `sector` and `borderline`")
+  }
```
Comment (maurolepore) [inline R/restructure_ald.R:78]: This error should only face maintainers -- as this condition is only met in testing environments where we allow reserved columns.

**Comment by maurolepore** [comment]
One test in CI throws an error but seems beyond our control
```
Error: Error: getaddrinfo ENOTFOUND rversions.r-pkg.org
```

**Review by jdhoffa — APPROVED** [review]
looketh goodeth to meh

**Pull request merged**